### DISCUSSION TITLE
how do i add error handling

### DISCUSSION THREAD

**Discussion by eurichard** [Rapptz/discord.py#9209] [Q&A]
so i know how to add error handling to discord.py with like prefix commands but im new to slash commands i dont know how can you help?

**Comment by LostLuma** [comment]
Hello, there are multiple ways to add error handling for app commands depending on how you create them, and which style you'd like to use.

For regular `app_command` package usage there are two ways of defining error handlers, either using the `@error` decorator on `CommandTree`:
```py
client = discord.Client(...)
tree = discord.app_commands.CommandTree(client)


@tree.error
async def on_error(interaction: discord.Interaction, error: discord.app_commands.AppCommandError) -> None:
    ...
```

Or alternatively you can subclass `Commandtree` and overwrite the `on_error` method with your own implementation:
```py

class MyCommandTree(discord.app_commands.CommandTree):
    async def on_error(self, interaction: discord.Interaction, error: discord.app_commands.AppCommandError) -> None:
        ...


client = discord.Client(...)
tree = MyCommandTree(client)
```

If you're using [hybrid commands](https://discordpy.readthedocs.io/en/latest/ext/commands/commands.html#hybrid-commands) your regular prefix command error handler will be used for app command errors as well.

Hope this helps!
